src/c_api/eigensolver/eigensolver.h

```diff
@@ -0,0 +1,154 @@
+    }  // Destroy mirror
+
+    // Ensure data is copied back to the host
+    eigenvalues_host.waitLocalTiles();
+    eigenvectors_host.waitLocalTiles();
+
+    pika::suspend();
+    return 0;
+  }
+  catch (const std::exception& e) {
+    std::cerr << e.what() << '\n';
+    return -1;
+  }
+  catch (...) {
+    std::cerr << "ERROR: Unknown exception caught in DLA-Future's eigensolver." << '\n';
+    return -1;
+  }
+}
+
+#ifdef DLAF_WITH_SCALAPACK
+
+template <typename T>
+void pxxxevd(char uplo, int m, T* a, [[maybe_unused]] int ia, [[maybe_unused]] int ja, int* desca,
+             dlaf::BaseType<T>* w, T* z, [[maybe_unused]] int iz, [[maybe_unused]] int jz, int* descz,
+             int& info) {
+  try {
+    using MatrixHost = dlaf::matrix::Matrix<T, dlaf::Device::CPU>;
+    using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
+    using MatrixBaseMirror =
+        dlaf::matrix::MatrixMirror<dlaf::BaseType<T>, dlaf::Device::Default, dlaf::Device::CPU>;
+
+    DLAF_ASSERT(desca[0] == 1, desca[0]);
+    DLAF_ASSERT(descz[0] == 1, descz[0]);
+    DLAF_ASSERT(desca[1] == descz[1], desca[1], descz[1]);
+    DLAF_ASSERT(ia == 1, ia);
+    DLAF_ASSERT(ja == 1, ja);
+    DLAF_ASSERT(iz == 1, iz);
+    DLAF_ASSERT(iz == 1, iz);
+
+    pika::resume();
+
+    // Get grid corresponding to blacs context in desca
+    // The grid needs to be created with dlaf_create_grid_from_blacs
+    auto communicator_grid = dlaf_grids.at(desca[1]);
+    dlaf::matrix::Distribution distribution_a({m, m}, {desca[4], desca[5]}, communicator_grid.size(),
+                                              communicator_grid.rank(), {desca[6], desca[7]});
+    dlaf::matrix::Distribution distribution_z({m, m}, {descz[4], descz[5]}, communicator_grid.size(),
+                                              communicator_grid.rank(), {descz[6], descz[7]});
+
+    dlaf::matrix::LayoutInfo layout_a = colMajorLayout(distribution_a, desca[8]);
+    dlaf::matrix::LayoutInfo layout_z = colMajorLayout(distribution_z, descz[8]);
+
+    MatrixHost matrix_host(distribution_a, layout_a, a);
+    MatrixHost eigenvectors_host(distribution_z, layout_z, z);
+    auto eigenvalues_host = dlaf::matrix::createMatrixFromColMajor<dlaf::Device::CPU>(
+        {m, 1}, {distribution_z.blockSize().rows(), 1}, std::max(m, 1), w);
+
+    {
+      MatrixMirror matrix(matrix_host);
+      MatrixMirror eigenvectors(eigenvectors_host);
+      MatrixBaseMirror eigenvalues(eigenvalues_host);
+
+      dlaf::eigensolver::eigensolver<dlaf::Backend::Default, dlaf::Device::Default, T>(
+          communicator_grid, blas::char2uplo(uplo), matrix.get(), eigenvalues.get(), eigenvectors.get());
+    }  // Destroy mirror
+
+    // Ensure data is copied back to the host
+    eigenvalues_host.waitLocalTiles();
+    eigenvectors_host.waitLocalTiles();
+
+    pika::suspend();
+
+    info = 0;
+  }
+  catch (const std::exception& e) {
+    std::cerr << e.what() << '\n';
+    info = -1;
```

Comment:
As above about exceptions.
As `info = -1` in lapack and scalapack means that the first function argument is wrong, is this the exception you want to catch?